src/server/pps/server/api_server.go

```diff
@@ -2607,8 +2607,20 @@ func (a *apiServer) CreatePipeline(ctx context.Context, request *pps.CreatePipel
 			return
 		}
 		if input.Pfs != nil && input.Pfs.Trigger != nil {
+			var prevHead *pfs.Commit
+			_, err := pfsClient.InspectBranch(ctx, &pfs.InspectBranchRequest{
+				Branch: client.NewBranch(input.Pfs.Repo, input.Pfs.Branch),
+			})
+
+			if err != nil && !isNotFoundErr(err) {
+				visitErr = err
+				return
+			} else if err == nil {
+				prevHead = client.NewCommit(input.Pfs.Repo, input.Pfs.Branch)
+			}
```

Comment:
kind of silly to critique my own code, but I guess I didn't really need the extra `err` variable

Reply:
I think you do, or you would need to clear it if `isNotFoundErr(err)`